File: shared/ocr_extractor.py
from PIL import Image
import pytesseract

def extract_text_from_image(image_path):
    try:
        text = pytesseract.image_to_string(Image.open(image_path), lang='rus+eng')
        return text.strip()
    except Exception as e:
        print(f"❌ Ошибка OCR: {e}")
        return ''

File: shared/ocr_extractor.js
const Tesseract = require('tesseract.js');

async function extractTextFromImage(imagePath) {
    try {
        const { data: { text } } = await Tesseract.recognize(imagePath, 'rus+eng');
        return text.trim();
    } catch (err) {
        console.error("❌ OCR ошибка:", err.message);
        return '';
    }
}

module.exports = { extractTextFromImage };